build/Dockerfile

```diff
@@ -1,28 +1,26 @@
-FROM --platform=linux/amd64 python:3.10-slim
+FROM --platform=linux/amd64 python:3.10-slim as builder
 
-RUN set -ex && pip install --upgrade pip && pip install poetry pipx
+RUN set -ex && pip install --upgrade pip && pip install poetry
 ENV PATH /root/.local/bin:$PATH
 
 RUN apt update
-RUN apt install -y curl
-RUN apt install -y less
-RUN curl -sL https://aka.ms/InstallAzureCLIDeb | bash
 
 RUN mkdir -p cdf_fabric_replicator 
 WORKDIR cdf_fabric_replicator
 
 COPY pyproject.toml ./
 COPY poetry.lock ./
 COPY cdf_fabric_replicator cdf_fabric_replicator
-COPY build/entrypoint.sh /
-RUN chmod +x /entrypoint.sh
-
-RUN pipx install .
 
 RUN mkdir -p /config
 COPY build/config_remote.yaml /config/config_remote.yaml
 
 WORKDIR cdf_fabric_replicator
+RUN poetry config virtualenvs.create false
+RUN poetry install --without dev
+
+FROM builder as base
+COPY cdf_fabric_replicator cdf_fabric_replicator
 
-ENTRYPOINT [ "/entrypoint.sh" ]
-CMD ["/config/config_remote.yaml"]
+WORKDIR cdf_fabric_replicator
+CMD ["python", "-m", "cdf_fabric_replicator", "/config/config_remote.yaml"]
```

Comment:
Good start! Let's touch base when you're back from vacation. I have some ideas on how to get the container even smaller.

Reply:
I was able to get the image down to 489MB, it may be challenging to get it reduced further as our project dependencies from pip are 362mb, pyarrow alone is 128mb